src/i_system.c

```diff
@@ -138,13 +136,12 @@ static exit_priority_t exit_priority;
 void I_AtExitPrio(atexit_func_t func, boolean run_on_error,
                   const char *name, exit_priority_t priority)
 {
-    atexit_listentry_t entry;
-
-    entry.func = func;
-    entry.run_on_error = run_on_error;
-    entry.name = name;
-
-    array_push(exit_funcs[priority], entry);
```

Comment:
This construction is too much for m_array; it will break when `realloc` is called.

Reply:
By the way, we could create an arena-based m_array that never invalidates pointers. I thought about it, but it feels too different from usual C.